debian/tests/mount

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/python3
+
+"""Mount test for official models.
+
+The official models are here:
+
+    http://people.canonical.com/~vorlon/official-models/
+
+For each of these, build an actual disk image.  For each image, go through the
+mountable partitions as defined in the gadget.yaml (e.g. not the mbr, but
+including the unspecified last writable partition), and make sure each
+partition can actually be mounted.
+"""
+
+import os
+import sys
+
+from contextlib import ExitStack
+from subprocess import PIPE, run
+from ubuntu_image.parser import FileSystemType, parse
+
+
+TMP = os.environ['AUTOPKGTEST_TMP']
+DIR = os.path.abspath(os.path.join('debian', 'tests', 'models'))
+STATUS = {}
+
+
+for model_file in os.listdir(DIR):
+    model_status = STATUS.setdefault(model_file, {})
+    print('==========>', model_file, flush=True)
+    base, suffix = os.path.splitext(model_file)
+    if suffix != '.assertion':
+        continue
+    # Build the image.
+    workdir = os.path.join(TMP, base)
+    model_assertion = os.path.join(DIR, model_file)
+    run(['ubuntu-image', '-c', 'edge', '-w', workdir, model_assertion])
+    # Now try to mount every mountable partition in the resulting disk image.
+    # First, we have to figure out which are the gadget-specified mountable
+    # partitions.  These are any defined with `filesystem: ext4|vfat`.
+    gadget_yaml = os.path.join(workdir, 'gadget.yaml')
+    with open(gadget_yaml, 'r', encoding='utf-8') as fp:
+        gadget = parse(fp.read())
```

Comment:
This of course means that these tests must rely on the correctness of the gadget yaml parser, since we use it both in the test and the actual code in order to define the "right" outcome.  I think this is ok, we just have to enforce the correctness of the parser itself one layer down in the unit tests (which should already be the case).